src/main/java/net/datafaker/transformations/JavaObjectTransformer.java

```diff
@@ -23,7 +25,32 @@ public Object apply(Object input, Schema<Object, ?> schema) {
         if (input instanceof Class) {
             clazz = (Class) input;
             result = CLASS2RESULT.get(clazz);
-            if (result == null) {
+        } else {
+            clazz = input.getClass();
+            result = input;
+        }
+
+        if (result == null) {
+            if (clazz.isRecord()) {
+                Class<?>[] componentTypes = Arrays.stream(clazz.getRecordComponents())
+                    .map(RecordComponent::getType)
+                    .toArray(Class<?>[]::new);
+
+                final Field<Object, ?>[] fields = schema.getFields();
+                final Object[] values = new Object[fields.length];
+                for (int i = 0; i < fields.length; i++) {
+                    values[i] = fields[i].transform(result);
+                }
+
+                try {
+                    result = clazz.getDeclaredConstructor(componentTypes).newInstance(values);
```

Comment:
it's better to cache it as well to avoid constant calculation logic

Reply:
I don't it's possible to cache object, but I can cache constructor. WDYT?

Reply:
yes, constructor should be enough

Reply:
@snuyanzin I have a question for you, imagine an article about Datafaker 2.0, what would you expect to see in that article, I mean topics or something like this

Reply:
depends on the target audience...

there is a list of topics missed in faker related talks

1. custom providers 
2. there are ways to generate sequences (stream and sequence) 
3. the transformation approach (csv, json, sql, java, custom)
4. object population as a replacement for builders

Reply:
Could I send you my draft of the article about Datafaker 2.0? I would like to hear your opinion if you don't mind and you have time, of course

Reply:
yep, sure